internal/rbd/nodeserver.go

```diff
@@ -1190,26 +1197,53 @@ func (ns *NodeServer) xfsSupportsReflink() bool {
 func (ns *NodeServer) NodeGetVolumeStats(
 	ctx context.Context,
 	req *csi.NodeGetVolumeStatsRequest) (*csi.NodeGetVolumeStatsResponse, error) {
-	var err error
-	targetPath := req.GetVolumePath()
-	if targetPath == "" {
-		err = fmt.Errorf("targetpath %v is empty", targetPath)
+	var (
+		imgInfo rbdImageMetadataStash
+		err     error
+	)
+	volumePath := req.GetStagingTargetPath()
+	if volumePath == "" {
+		if volumePath = req.GetVolumePath(); volumePath == "" {
+			return nil, status.Error(codes.InvalidArgument, "volume path must be provided")
+		}
+	} else {
+		imgInfo, err = lookupRBDImageMetadataStash(volumePath)
+		if err != nil {
+			log.ErrorLog(ctx, "failed to find image metadata, at stagingPath: %s, err: %v", volumePath, err)
 
-		return nil, status.Error(codes.InvalidArgument, err.Error())
+			return nil, status.Error(codes.NotFound, err.Error())
+		}
 	}
 
-	stat, err := os.Stat(targetPath)
+	var volCondition *csi.VolumeCondition
+	stat, err := os.Stat(volumePath)
 	if err != nil {
-		return nil, status.Errorf(codes.InvalidArgument, "failed to get stat for targetpath %q: %v", targetPath, err)
+		volCondition.Abnormal = true
+		volCondition.Message = fmt.Sprintf("failed to get stat for volumepath %q: %v", volumePath, err)
+	}
+
+	// userspace mounters
+	if imgInfo.NbdAccess {
+		stdout, stderr, err := util.ExecCommand(ctx, "ps", "aux", "|", "grep", "[r]bd-nbd")
```

Comment:
is there any simplification for this one? if we have more nbd mounts this will return a big list of mounts?

Reply:
@pkalever, not all base images we use for cephcsi will have `ps` installed by default (downstream)?

Reply:
We depend on many such utilities and I would expect it to be present. I think `ps` is part of `procps-ng` package, can we fix it with the container images?

In case `ps` is not available then we need to search all the `/proc/*/cmdline` one at a time and that would be time-consuming. 

There is also another way to do this via `rbd-nbd list-mapped` command, but maybe lets conclude if the unavailability of `ps` is acceptable for us, because maybe others might use it (if not now maybe in the future)

cc: @nixpanic

Reply:
we don't have `ps` package in downstream cephcsi containers. if we want it we need to make it as required.

```
[🎩︎]mrajanna@fedora logs $]oc exec -it csi-rbdplugin-provisioner-58995c7c7b-hrchr -c csi-rbdplugin -- sh
sh-4.4# ps
sh: ps: command not found
sh-4.4#
```

Reply:
Ok, let's then mark it as required?

Reply:
Using pstree for the time being, lets see what others got to say.